### ISSUE TITLE
Add the ability to use view rotation in style expressions.

### ISSUE THREAD

**Issue by syntax42** [openlayers/openlayers#15216]
I am looking to implement a website, where the rotation of the map has an impact on the styling of features in a webgl vector layer. As of now, the resolution and zoom of the view is present as a style expression, but not rotation.

I have made a branch implementing this, and i can create a PR if approved.

**Comment by ahocevar** [comment]
Are your use cases beyond rotating symbols or text with the view or leaving them upright?

**Comment by syntax42** [comment]
Yes @ahocevar ! My specific use case is to implement a "graticule border" that supports rotation. I have implemented a version of it here: https://historiskatlas6.azurewebsites.net/

If you try to rotate the map (alt + shift + mouse drag), you can se the border reacting to the change. This is implemented using a several webgl vector (polygon) layers and using gl.scissor between the layers. Notice what happens when you rotate the view beyond 45 degrees. The border markings switches between showing latitudes and longitudes. This effect is only possible at the moment by using several webgl layers. The performance of this is not good, so i wish to implement a solution using only one layer. This is where the rotation would be very valuable to have in a style expression.

I think i found a way around this though, but it involves using custom shaders on the webgl vector layer, and as far as i can see, this is not possible in version 8.1.0. This [commit](https://github.com/openlayers/openlayers/commit/afe990d3cb5cfc0d424f44417742d5e5474ab9ea) seems to fix that issue, but i do not know when this will roll out?

**Comment by jahow** [comment]
Nice use case! This sounds very easy to implement. I'll mark that as PR accepted!

Note that you cannot provide custom shaders anymore, but the `ShaderBuilder` class should offer you enough flexibility to achieve your goals. If not, I'd be interested in hearing what limitations you encountered.

**Comment by syntax42** [comment]
Thanks @jahow ! I had a look at `ShaderBuilder`, and it sure looks like it would do the job. Maybe i should just wait for the next release that includes det possibility to use `ShaderBuilder` on a `VectorLayer`?

**Comment by jahow** [comment]
That's right, using the ShaderBuilder instead of custom shaders will be part of the next release. You can also already experiment with the `ol@dev` package on NPM.
